### PR TITLE
Add ThreadSafe annotation to com.bugsnag.android, remove infer dependency

### LINKED ISSUE #366: infer-annotation bundled in SDK
Our CI environment highlighted the following transitive dependency being pulled in to our build:

```@@ -249,2 +249,4 @@
-+--- com.bugsnag:bugsnag-android:4.5.0
-|    \--- com.android.support:support-annotations:27.0.0 -> 27.1.1
++--- com.bugsnag:bugsnag-android:4.6.0
+|    +--- com.android.support:support-annotations:27.0.0 -> 27.1.1
+|    \--- com.facebook.infer.annotation:infer-annotation:0.11.2
+|         \--- com.google.code.findbugs:jsr305:3.0.1
```

Here's the line in question:
https://github.com/bugsnag/bugsnag-android/blob/0200a0e6dd21a0667b77d3762de44b83b0ce7058/sdk/build.gradle#L68

Since Infer is a static analyzer tool, might it be better to add it as a buildscript dependency or use a plugin like: https://github.com/uber-common/infer-plugin?

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal
Removes the need for a compile-time dependency on the infer-annotation artefact.

## Changeset
- Added `@ThreadSafe` annotation to bugsnag package
- Removed imports of infer-annotation 
- Altered build process so that the `infer` property must be set to compile infer-annotation artefact

## Tests
- Ran `./infer.sh`
- Ran `./gradlew sdk:assemble` to confirm
- Ran `generatePomFileForProductionPublication` task to confirm that the generated POM excludes the infer-annotation dependency.
- Ran existing unit/mazerunner tests on CI

### Alternative Approaches

We could attempt forking Uber's [infer plugin](https://github.com/uber-common/infer-plugin), which is deprecated.

### Linked issues

Fixes #366 

## Review

For the submitter, initial self-review:

- [x] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [x] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [x] Initial review of the intended approach, not yet feature complete
  - [x] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: infer.sh [hunk 1]
```diff
@@ -1,3 +1,5 @@
 #!/usr/bin/env bash
 ./gradlew clean && infer --disable-issue-type "NULL_DEREFERENCE" \
---disable-issue-type "RESOURCE_LEAK" -- ./gradlew sdk:build
+--disable-issue-type "RESOURCE_LEAK" \
+--threadsafe-aliases "[\"ThreadSafe\"]" \
```
Comment (fractalwrench) [inline infer.sh]: adds `com.bugsnag.android.ThreadSafe` as an alias for `com.facebook.infer.annotation.ThreadSafe`
